From 7e59f4a3425980304958c1575fd283be3b1a7830 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Sat, 3 Mar 2018 19:36:52 +0100
Subject: [PATCH 132/249] wiki: Add more validation

* Adds custom_twrp_link, download_boot, models and recovery_boot - all
  optional
* Adds tree - required

Change-Id: Ic2b343e033213ba04866200afb1bfa060c99d17b
---
 _data/devices/P024.yml  |  2 +-
 _data/devices/kiwi.yml  |  3 ++-
 _data/devices/m7.yml    |  2 +-
 _data/devices/m7vzw.yml |  2 +-
 _data/devices/n5110.yml |  2 +-
 _data/devices/n5120.yml |  2 +-
 _data/devices/v500.yml  |  2 +-
 test/schema-06.yml      | 18 ++++++++++++++++++
 8 files changed, 26 insertions(+), 7 deletions(-)

diff --git a/_data/devices/P024.yml b/_data/devices/P024.yml
index eb529e6..f804707 100644
--- a/_data/devices/P024.yml
+++ b/_data/devices/P024.yml
@@ -24,7 +24,7 @@ network:
 - {tech: 4G, bands: '1(2100) 2(1900) 3(1800) 5(850) 7(2600) 8(900) 20(800) LTE' }
 peripherals: [Accelerometer, Proximity sensor, Compass, GPS]
 ram: 1/2 GB
-recovery_boot:
+recovery_boot: ''
 release: 2015-07
 screen: 203 mm (8.0 in)
 screen_ppi: ~189
diff --git a/_data/devices/kiwi.yml b/_data/devices/kiwi.yml
index f11b001..5c114b0 100644
--- a/_data/devices/kiwi.yml
+++ b/_data/devices/kiwi.yml
@@ -12,7 +12,8 @@ cpu_cores: '8'
 cpu_freq: 4 x 1.5 GHz + 4 x 1.2 GHz
 current_branch: 14.1
 depth: 8.2 mm (0.32 in)
-download_boot:
+download_boot: With the device powered off, hold <kbd>Volume Down</kbd> + <kbd>Power</kbd>.
+  Keep holding buttons until the "Honor" logo appears, then release all buttons.
 gpu: Adreno 405
 height: 151.3 mm (5.96 in)
 image: kiwi.png
diff --git a/_data/devices/m7.yml b/_data/devices/m7.yml
index efe0f78..7a57679 100644
--- a/_data/devices/m7.yml
+++ b/_data/devices/m7.yml
@@ -12,7 +12,7 @@ cpu_cores: '4'
 cpu_freq: 1.7 GHz
 current_branch: 14.1
 depth: 9.3 mm
-download_boot:
+download_boot: ''
 gpu: Adreno 320
 height: 137.4 mm
 image: m7.png
diff --git a/_data/devices/m7vzw.yml b/_data/devices/m7vzw.yml
index ff70e71..e3a924f 100644
--- a/_data/devices/m7vzw.yml
+++ b/_data/devices/m7vzw.yml
@@ -12,7 +12,7 @@ cpu_cores: '4'
 cpu_freq: 1.7 GHz
 current_branch: 14.1
 depth: 9.3 mm
-download_boot:
+download_boot: ''
 gpu: Adreno 320
 height: 137.4 mm
 image: m7.png
diff --git a/_data/devices/n5110.yml b/_data/devices/n5110.yml
index 459004a..cf363d0 100644
--- a/_data/devices/n5110.yml
+++ b/_data/devices/n5110.yml
@@ -19,7 +19,7 @@ image: n5100.png
 install_method: heimdall
 kernel: android_kernel_samsung_smdk4412
 maintainers: [rgib]
-models:
+models: []
 name: Galaxy Note 8.0 (Wi-Fi)
 network:
 - {tech: 2G, bands: '' }
diff --git a/_data/devices/n5120.yml b/_data/devices/n5120.yml
index 724bfc4..11ddf77 100644
--- a/_data/devices/n5120.yml
+++ b/_data/devices/n5120.yml
@@ -19,7 +19,7 @@ image: n5100.png
 install_method: heimdall
 kernel: android_kernel_samsung_smdk4412
 maintainers: [rgib]
-models:
+models: []
 name: Galaxy Note 8.0 (LTE)
 network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
diff --git a/_data/devices/v500.yml b/_data/devices/v500.yml
index 72d70f0..66cd482 100644
--- a/_data/devices/v500.yml
+++ b/_data/devices/v500.yml
@@ -12,7 +12,7 @@ cpu_cores: '4'
 cpu_freq: 1.7 GHz
 current_branch: 14.1
 depth: 8.3 mm (0.33 in)
-download_boot:
+download_boot: ''
 gpu: Adreno 320
 height: 216.8 mm (8.54 in)
 image: v500.png
diff --git a/test/schema-06.yml b/test/schema-06.yml
index b673496..3eb507c 100644
--- a/test/schema-06.yml
+++ b/test/schema-06.yml
@@ -212,10 +212,17 @@ properties:
   current_branch:
     "$id": "/properties/current_branch"
     "$ref": "#/definitions/valid_branches"
+  custom_twrp_link:
+    "$id": "/properties/custom_twrp_link"
+    type: string
+    pattern: ^https?:\/\/(.*)
   custom_unlock_cmd:
     "$id": "/properties/custom_unlock_cmd"
     minLength: 10
     type: string
+  download_boot:
+    "$id": "/properties/download_boot"
+    type: string
   gpu:
     "$id": "/properties/gpu"
     enum:
@@ -281,6 +288,9 @@ properties:
     "$id": "/properties/maintainers"
     minItems: 0
     type: array
+  models:
+    "$id": "/properties/models"
+    type: array
   name:
     "$id": "/properties/name"
     minLength: 1
@@ -297,6 +307,9 @@ properties:
       type: array
     - enum:
       - None
+  recovery_boot:
+    "$id": "/properties/recovery_boot"
+    type: string
   release:
     "$id": "/properties/release"
     description: Date device released to public in ISO8601 reduced precision format
@@ -316,6 +329,10 @@ properties:
     "$id": "/properties/required_bootloader"
     minItems: 1
     type: array
+  tree:
+    "$id": "/properties/tree"
+    pattern: ^android_device_(.*)$
+    type: string
   type:
     "$id": "/properties/type"
     enum:
@@ -405,6 +422,7 @@ required:
 - name
 - peripherals
 - release
+- tree
 - type
 - vendor
 - vendor_short
-- 
2.17.1

